### ISSUE TITLE
`vmq-admin session show` not showing any sessions

### ISSUE THREAD

**Issue by trehn** [erlio/vernemq#526]
### Environment

- VerneMQ Version: 1.2.0
- OS: Docker (official image)

### Expected behavior

`docker exec -it XXX vmq-admin session show` should list current sessions.

### Actual behaviour

The command has no output (exit code 0). 

There is a client connected however:

```
docker exec -it XXX vmq-admin trace client client-id=XXX
Starting trace for 1 existing sessions for client "XXX" with PIDs
    [<7244.462.0>]
<7244.462.0> MQTT RECV: CID: "XXX" PINGREQ()
<7244.462.0> MQTT SEND: CID: "XXX" PINGRESP()
```

Trying to get the sessions over the HTTP API (GET /api/v1/sessions) just produces an HTTP 400 error and this log message:

```
[error] <0.500.0>@vmq_http_mgmt_api:malformed_request:79 malformed request {no_matching_spec,["vmq-admin","session","list","--format=json"]}
```

Not sure why it says "list" there instead of "show" (as --help suggests).

**Comment by ioolkos** [comment]
Hi @trehn thanks for pointing this out.
We tried to streamline the wording of the vmq-admin stuff (that explains why it was 'list' before). And it was an oversight to not adapt this in the HTTP API (or the logging at least).

`vmq-admin session show` should work though. vmq-admin starts a temporary Erlang node on the machine where Verne is running. So it has to be called there. Which is what you probably do.

Anyone seeing more help I can provide here? docker problem?

**Comment by larshesel** [comment]
Not sure what it is but can confirm that plain `vmq-admin session show` returns something but the http interface does not. We'd have to look into why (first step I think would be to see if it is reproducible outside docker).

**Comment by larshesel** [comment]
Just tested this outside a docker container and there it works - so this has something to do with the docker environment.

**Comment by larshesel** [comment]
Just to be summarize and keep the issues separate:

1. The call to `/api/v1/sessions` doesn't work as this is an alias for `/api/v1/session/list` which should be `/api/v1/session/show` as `list` was deprecated for `show` before 1.0.0 as @ioolkos correctly mentions. So this is a bug which I've created a PR for ( #530 ).

2. When using docker, calling `/api/v2/session/show` gives an error. I've narrowed it down to the following. When calling `session show` inside the docker image the 'peer_host' field is reported as `{172,17,0,1}` (or something similar), but outside docker (at least on my machine) it gets reported as a real name, in my case `localhost`. So it would seem our json encoder is not able to encode the string `{172,17,0,1}` correctly and therefore the request crashes.

You can work around this by calling `/api/v1/sessions\?--client_id\&--user\&--is_online` leaving out the problematic `peer_host` field.

**Comment by trehn** [comment]
`/api/v1/session/show` "works":

```
> curl -i http://XXXXX@localhost:8082/api/v1/session/show\?--client_id\&--user\&--is_online
HTTP/1.1 200 OK
server: Cowboy
date: Fri, 20 Oct 2017 15:07:25 GMT
content-length: 27
content-type: application/json

{"table":[],"type":"table"}
```

I say "works" because the table shouldn't be empty, but at least it's there ;)

**Comment by trehn** [comment]
Every time I fetch the empty table over HTTP, roughly a second after the request completes, I get this log message:

```
[warning] <0.551.0>@vmq_ql_query:select:197 Subquery failed due to timeout
```

Probably somehow related to why I'm not seeing any sessions.

**Comment by codeadict** [comment]
I can test that both inside and outside docker, looks like a good bug to work on for me.

**Comment by codeadict** [comment]
Can confirm it working outside docker on Mac
![screen shot 2017-10-23 at 4 13 14 pm](https://user-images.githubusercontent.com/168440/31910965-2dc1817c-b80d-11e7-84f5-177adba54f8c.png)

**Comment by codeadict** [comment]
Outside docker works fine
![screen shot 2017-10-23 at 4 14 34 pm](https://user-images.githubusercontent.com/168440/31911011-4a08d240-b80d-11e7-8294-d4c94c229795.png)

**Comment by larshesel** [comment]
I've created a fix for the `peer_host` encoding problem in #531 

@trehn do you still have the issue with the timeouts?

**Comment by trehn** [comment]
There doesn't seem to be an easy way for me to test the fix inside Docker (like a nightly build or something).

Were you able to reproduce the problem inside Docker?

(I'll be unavailable now until Nov 2nd)

**Comment by larshesel** [comment]
The peer_host fix is being released as 1.2.1 and docker images will be published soon. It would be a great help if you could test these and see if you run into any issues.

**Comment by trehn** [comment]
It kinda works now, tracking remaining issue in #542.
